upf/upf_cli.c

```diff
@@ -352,9 +355,12 @@ upf_nat_pool_add_del_command_fn (vlib_main_t * vm,
 	error = clib_error_return (0, "Unable to create NAT Pool");
     }
 
+done:
+  vec_free (nwi_name);
+  vec_free (name);
   vec_free (nwi_s);
+  unformat_free (line_input);
```

Comment:
This will crash if `unformat_user` returned 0
See
https://github.com/FDio/vpp/blob/2652867480a185ccdeccf9ca9210a1213b82cf26/src/vppinfra/format.h#L155-L161
https://github.com/FDio/vpp/blob/2652867480a185ccdeccf9ca9210a1213b82cf26/src/vppinfra/unformat.c#L398-L407
this happens when an empty input (rest of the command) is encountered
https://github.com/FDio/vpp/blob/2652867480a185ccdeccf9ca9210a1213b82cf26/src/vppinfra/unformat.c#L381-L395